src/main/java/Cart.java

```diff
@@ -0,0 +1,80 @@
+import exceptions.NotFoundInArrayList;
+
+import java.util.ArrayList;
+
+public class Cart {
+    private ArrayList<CartItem> items = new ArrayList<>();
+    private int deliveryFee;
+    
+   public void setDeliveryFee(int deliveryFee) {
+        this.deliveryFee = deliveryFee;
+    }
+
+    public int getDeliveryFee() {
+        return deliveryFee;
+    }
+    
+    public Cart(){
+    }
+
+    //Calculate the VAT
+    private Double calculateVAT(double itemFullPrice){
+        return itemFullPrice * 0.2 ;
+    }
+
+    //Calculate the final price of all products
+    private Double calculateFinalPrice(){
+        double sum = 0;
+        for (var item: items) {
+            sum += item.calculateTheFinalSumWithQuantity(item.getQuantity());
+        }
+        if(sum > 0) {
+            if (sum < 100) {
+                setDeliveryFee(10);
+            } else if (sum < 200) {
+                setDeliveryFee(5);
+            } else {
+                setDeliveryFee(0);
+            }
+        }else{
+            System.out.println("Your cart is empty. Please add an item to calculate!");
+        }
+        return (sum + getDeliveryFee()) - calculateVAT(sum);
+    }
+
+    //Add the item to the cart
+    public void addItemToCart(CartItem cartItem){
+        items.add(cartItem);
+    }
+
+    /**
+     * @variable item: searches in the cartItems ArrayList the cartItem object which is gived as parameter.
+     * @variable itemQuantity:get the quantity of the current CartItem object
+     * Also, if the quantity is below than zero the item will be removed from the ArrayList.
+     * Finally, if we call the method below again for the item which is already deleted it, it throws sample custom exception.
+     */
+    public void deleteItemFromCart(CartItem cartItem) throws NotFoundInArrayList {
+        try {
+            var item =items.stream().filter(x-> x.getProduct().getLabel() == cartItem.getProduct().getLabel()).findAny();
```

Comment:
When using method chaining in function style, move the calls in the chain in a new line to make it readable